redfish-core/lib/hypervisor_system.hpp

```diff
@@ -311,247 +433,370 @@ template <typename CallbackFunc>
+    crow::connections::systemBus->async_method_call(
+        [asyncResp, ifaceId](const boost::system::error_code& ec) {
         if (ec)
         {
-            BMCWEB_LOG_ERROR("DBUS response error {}", ec);
-            return;
-        }
-        BMCWEB_LOG_DEBUG("SubnetMask is Set");
-    });
-}
-
-/**
- * @brief Sets the Hypervisor Interface Gateway DBUS
- *
- * @param[in] asyncResp          Shared pointer for generating response message.
- * @param[in] gateway        Gateway from the incoming request
- * @param[in] ethIfaceId     Hypervisor Interface Id
- *
- * @return None.
- */
-inline void setHypervisorIPv4Gateway(
-    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
-    const std::string& gateway)
-{
-    BMCWEB_LOG_DEBUG(
-        "Setting the DefaultGateway to the last configured gateway");
-
-    sdbusplus::asio::setProperty(
-        *crow::connections::systemBus, "xyz.openbmc_project.Settings",
-        "/xyz/openbmc_project/network/hypervisor",
-        "xyz.openbmc_project.Network.SystemConfiguration", "DefaultGateway",
-        gateway, [asyncResp](const boost::system::error_code& ec) {
-        if (ec)
-        {
-            BMCWEB_LOG_ERROR("DBUS response error {}", ec);
+            messages::internalError(asyncResp->res);
             return;
         }
-        BMCWEB_LOG_DEBUG("Default Gateway is Set");
-    });
-}
-
-/**
- * @brief Creates a static IPv4 entry
- *
- * @param[in] ifaceId      Id of interface upon which to create the IPv4 entry
- * @param[in] prefixLength IPv4 prefix syntax for the subnet mask
- * @param[in] gateway      IPv4 address of this interfaces gateway
- * @param[in] address      IPv4 address to assign to this interface
- * @param[io] asyncResp    Response object that will be returned to client
- *
- * @return None
- */
-inline void
-    createHypervisorIPv4(const std::string& ifaceId, uint8_t prefixLength,
-                         const std::string& gateway, const std::string& address,
-                         const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
-{
-    setHypervisorIPv4Address(asyncResp, ifaceId, address);
-    setHypervisorIPv4Gateway(asyncResp, gateway);
-    setHypervisorIPv4Subnet(asyncResp, ifaceId, prefixLength);
-}
-
-/**
- * @brief Deletes given IPv4 interface
- *
- * @param[in] ifaceId     Id of interface whose IP should be deleted
- * @param[io] asyncResp   Response object that will be returned to client
- *
- * @return None
- */
-inline void
-    deleteHypervisorIPv4(const std::string& ifaceId,
-                         const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
-{
-    std::string address = "0.0.0.0";
-    std::string gateway = "0.0.0.0";
-    const uint8_t prefixLength = 0;
-    setHypervisorIPv4Address(asyncResp, ifaceId, address);
-    setHypervisorIPv4Gateway(asyncResp, gateway);
-    setHypervisorIPv4Subnet(asyncResp, ifaceId, prefixLength);
+        std::string eventOrigin = std::format(
```

Comment:
Sorry I incorrectly suggested.

As it is URI, the proper one to use `boost::urls::format()' instead of `std::format()`.

```
std::string eventOrigin = boost::urls::format(...);
```

Reply:
done